auction/contracts/AuctionHouse.sol

```diff
@@ -0,0 +1,333 @@
+            _sellAsset.assetType.assetClass != LibAsset.ETH_ASSET_CLASS && 
+            _sellAsset.assetType.assetClass != LibAsset.ERC20_ASSET_CLASS,
+            "can't sell ETH or ERC20 on auction"
+        );
+
+        LibAucDataV1.DataV1 memory aucData = LibAucDataV1.parse(data, dataType);
+        require(aucData.duration >= EXTENSION_DURATION && aucData.duration <= MAX_DURATION, "incorrect duration");
+
+        uint endTime = 0;
+        if (aucData.startTime > 0){
+            require (aucData.startTime >= block.timestamp, "incorrect start time");
+            endTime = aucData.startTime + aucData.duration;
+        }
+        uint currentAuctionId = getNextAndIncrementAuctionId();
+        auctions[currentAuctionId] = Auction(
+            _sellAsset,
+            _buyAsset,
+            Bid(0, "", ""),
+            _msgSender(),
+            payable(address(0)),
+            endTime,
+            minimalStep,
+            minimalPrice,
+            protocolFee,
+            dataType,
+            data
+        );
+        transfer(_sellAsset, _msgSender(), address(this), TO_LOCK, LOCK);
+        setApproveForTransferProxy(_sellAsset);
+
+        setAuctionForToken(_sellAsset, currentAuctionId);
+        emit AuctionCreated(currentAuctionId, auctions[currentAuctionId]);
+    }
+
+    /// @dev sets approval for transfer proxy to transfer sell asset from this contract
+    function setApproveForTransferProxy(LibAsset.Asset memory _asset) internal {
+        if (_asset.assetType.assetClass == LibAsset.ERC20_ASSET_CLASS) {
+            (address token) = abi.decode(_asset.assetType.data, (address));
+            IERC20Upgradeable(token).approve(proxies[LibAsset.ERC20_ASSET_CLASS], _asset.value);
+        } else if (_asset.assetType.assetClass == LibAsset.ERC721_ASSET_CLASS) {
+            (address token,) = abi.decode(_asset.assetType.data, (address, uint256));
+            require(_asset.value == 1, "erc721 value error");
+            IERC721Upgradeable(token).setApprovalForAll(proxies[LibAsset.ERC721_ASSET_CLASS], true);
+        } else if (_asset.assetType.assetClass == LibAsset.ERC1155_ASSET_CLASS) {
+            (address token,) = abi.decode(_asset.assetType.data, (address, uint256));
+            IERC1155Upgradeable(token).setApprovalForAll(proxies[LibAsset.ERC1155_ASSET_CLASS], true);
+        }
+    }
+
+    /// @dev increments auctionId and returns new value
+    function getNextAndIncrementAuctionId() internal returns (uint256) {
+        return auctionId++;
+    }
+
+    /// @dev put a bid and return locked assets for the last bid
+    function putBid(uint _auctionId, Bid memory bid) payable public {
+        checkAuctionInProgress(_auctionId);
+        address payable newBuyer = _msgSender();
+        uint newAmount = bid.amount;
+        if (buyOutVerify(_auctionId, newAmount)) {
+            _buyOut(_auctionId, bid);
+            doTransfers(_auctionId);
+            deactivateAuction(_auctionId);
+            return;
```

Comment:
I would add here either emit BidPlaced or emit BuyOut (or similar to)
Because when somebody buys out auction we will have log only about "auctionFinished". I think it's important to know that it was bought out by logs or finished with previous bid.

the same problem is in buyOut method. so maybe it's better to add "emit event" in _buyOut method

Reply:
well, we had buyout event at first, but decided to get rid of it after thinking about it with back end team, we could probably make a call about it if you think that's needed